libctru/source/services/fs.c

```diff
@@ -9,20 +9,30 @@
 #include <3ds/env.h>
 #include "../internal.h"
 
+#define FS_MAX_ARCHIVE_HANDLES 32
```

Comment:
The point is that it is extremely rare to want to bypass the session for so many archives.